mockserver/data_reader.go

```diff
@@ -28,12 +28,73 @@ type BlockHeader struct {
 	NextBlock chainhash.Hash `json:"nextblockhash"`
 }
 
+type Transaction struct {
+	TxId          chainhash.Hash `json:"txid"`
+	Hash          chainhash.Hash `json:"hash"`
+	Version       int32          `json:"version"`
+	Size          int32          `json:"size"`
+	VSize         int32          `json:"vsize"`
+	Weight        int32          `json:"weight"`
+	LockTime      uint32         `json:"locktime"`
+	VIn           []Vin          `json:"vin"`
+	VOut          []Vout         `json:"vout"`
+	Hex           string         `json:"hex"`
+	BlockHash     chainhash.Hash `json:"blockhash"`
+	Confirmations uint32         `json:"confirmations"`
+	Time          uint32         `json:"time"`
+	BlockTime     uint32         `json:"blocktime"`
+}
+
+// Vout models parts of the tx data.  It is defined separately since both
+// getrawtransaction and decoderawtransaction use the same structure.
+type Vout struct {
+	Value        float64            `json:"value"`
```

Comment:
are you sure this is correct? I don't think it can be included in the block.
I found this types:

+ https://github.com/btcsuite/btcd/blob/master/wire/msgtx.go#L269
+ https://github.com/btcsuite/btcd/blob/master/wire/msgtx.go#L336

Reply:
Type removed.